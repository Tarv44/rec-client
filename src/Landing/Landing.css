.landing {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-around;
    margin: 0 2%;
}

.landing-title {
    text-transform: uppercase;
}

.welcome-section {
    border: #0CF0AB solid 3px;
    padding: 5%;
    margin: 3%;
}

.welcome-section h2 {
    border-bottom: solid #0CF0AB 1px;
    color: #454545;
}



.landing > * {
    min-height: 29vh;
}

.get-started {
    background-color: #454545;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: center;
    color: white;
}

.get-started h2 {
    border-bottom: #F0E118 solid 1px;
    margin: 0;
}



.signup-button {
    margin-right: 10px;
}

.account-buttons {
    text-align: center;
}