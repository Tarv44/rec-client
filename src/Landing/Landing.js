import { Component } from 'react'
import { NavLink } from 'react-router-dom'
import './Landing.css'

export default class Landing extends Component {
    render() {
        return (
            <main className='landing'>
                <header>
                    <h1 className='landing-title'>Rotation</h1>
                </header>

                <section>
                    <h2>Intro</h2>
                    <p>Introductory text that informs what the heck is going here on this site.</p>
                </section>

                <section>
                    <h2>Get Started</h2>
                    <p>Create an account or login to make a new exchange.</p>
                    <div className='account-buttons'>
                        <NavLink to={'/signup'} className='signup-button'><button>Signup</button></NavLink>
                        <NavLink to={'/login'}><button>Login</button></NavLink>
                    </div>
                </section>
            </main>
        )
    }
}