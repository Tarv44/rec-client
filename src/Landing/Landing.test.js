import React from 'react'
import ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import Landing from './Landing'
import { BrowserRouter } from 'react-router-dom'


describe('----- Landing -----', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render((
            <BrowserRouter>
                <Landing />
            </BrowserRouter> 
        ), div)
        ReactDom.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const comp = renderer
            .create(
                <BrowserRouter>
                    <Landing />
                </BrowserRouter> 
            )
            .toJSON()
        expect(comp).toMatchSnapshot()
    })
})