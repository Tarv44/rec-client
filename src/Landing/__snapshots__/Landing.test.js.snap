// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----- Landing ----- renders the UI as expected 1`] = `
<main
  className="landing"
>
  <header
    className="landing-header"
  >
    <h1
      className="landing-title"
    >
      Rotatio
      <span
        id="last-letter"
      >
        n
      </span>
    </h1>
  </header>
  <section
    className="welcome-section"
  >
    <h2>
      Welcome!
    </h2>
    <p>
      Whether you're a musician, a record head, or just an excited fan, it's a pleasure to share the music you care about with those around you. These days, "those around you" is likely just your roommate or dog.
    </p>
    <p>
      Rotation was created as a tool for deeper music conversation between acquaintances far and wide, that goes beyond texting a basic link to a streaming platform. It allows you to create your own chat room for the music you're listening to that you can share with your friends.
    </p>
  </section>
  <section
    className="get-started"
  >
    <h2>
      Get Started
    </h2>
    <p>
      Create an account (or login) and make a new exchange to get started. Once you've made an exchange, share the url for your new exchange with a friend to get them in on the convo.
    </p>
    <div
      className="account-buttons"
    >
      <a
        aria-current={null}
        className="signup-button"
        href="/signup"
        onClick={[Function]}
      >
        <button>
          Signup
        </button>
      </a>
      <a
        aria-current={null}
        href="/login"
        onClick={[Function]}
      >
        <button>
          Login
        </button>
      </a>
    </div>
  </section>
</main>
`;
